src/PresentationalComponents/Inventory/helpers.js

```diff
@@ -152,3 +157,100 @@ export const allCurrentSystemIds =
     setIsLoading(false);
     return merged;
   };
+
+export const pathwayCheck = async (
+  hasPathwayDetails,
+  pathway,
+  setHasPathwayDetails,
+  setPathwayReportList,
+  setPathwayRulesList
+) => {
+  if (!hasPathwayDetails) {
+    if (pathway) {
+      let pathwayRules = (
+        await Get(
+          `${BASE_URL}/pathway/${encodeURI(pathway.slug)}/rules/`,
+          {},
+          {}
+        )
+      )?.data.data;
+
+      let pathwayReport = (
+        await Get(
+          `${BASE_URL}/pathway/${encodeURI(pathway.slug)}/reports/`,
+          {},
+          {}
+        )
+      )?.data.rules;
+      setHasPathwayDetails(true);
+      setPathwayReportList(pathwayReport);
+      setPathwayRulesList(pathwayRules);
+    }
+  }
+};
+
+export const rulesCheck = async (
+  rule,
+  rulesPlaybookCount,
+  filters,
+  setRulesPlaybookCount
+) => {
+  if (rulesPlaybookCount < 0) {
+    const associatedRuleDetails = (
+      await Get(
+        `${RULES_FETCH_URL}${encodeURI(rule.rule_id)}/`,
+        {},
+        { name: filters.name }
+      )
+    )?.data.playbook_count;
+    setRulesPlaybookCount(associatedRuleDetails);
+  }
+};
+
+export const useRemediationButtonStatus = (
+  pathwayReportList,
+  selectedIds,
+  pathway,
+  pathwayRulesList,
+  rulesPlaybookCount
+) => {
+  let playbookFound = false;
+  const [isRemediationButtonDisabled, setIsRemediationButtonDisabled] =
+    useState(true);
+  let ruleKeys = Object.keys(pathwayReportList);
+
+  useEffect(() => {
+    if (selectedIds?.length <= 0 || selectedIds === undefined) {
+      setIsRemediationButtonDisabled(true);
+    } else if (pathway) {
+      for (let i = 0; i < selectedIds?.length; i++) {
+        let system = selectedIds[i];
+        if (playbookFound) {
+          break;
+        }
+        ruleKeys.forEach((rule) => {
```

Comment:
`forEach` doesn't seem right here. This could probably be done with filter/any, or?

Reply:
We want to check against each ruleKey. ruleKeys are used to see if the report list[ruleKey] has the system we are currently on. If reportList[ruleKey] doesn't have it, we need to move on to the next one. Typically there arent a lot of rule keys, but each needs to be looked through. using find() wouldnt allow for the checking of each rulekey against pathwayReportList. 
Thats why the find() is currently used on line 236. At least I dont yet see how to implement find() there

Reply:
I've been looking into this function, and maybe let's go over it together. I think we can make it way more efficient and clean.